### PR TITLE
test: Refactor integration test

### PR DESCRIPTION
**Pull request by rish691**
Move out all the common test for Postgres & Mongo

### CODE REVIEW

File: document-store/src/integrationTest/java/org/hypertrace/core/documentstore/mongo/MongoDocStoreTest.java [hunk 1]
```diff
@@ -124,141 +124,22 @@ public void testCollections() {
...
-        documents.add(results.next());
-      }
-      Assertions.assertFalse(documents.isEmpty());
-      String persistedDocument = documents.get(0).toJson();
-      JsonNode jsonNode = OBJECT_MAPPER.reader().readTree(persistedDocument);
-      Assertions.assertTrue(persistedDocument.contains("Bob"));
-      Assertions.assertTrue(jsonNode.findValue("createdTime").asLong(0) >= now);
-      Assertions.assertTrue(jsonNode.findValue("lastUpdatedTime").asLong(0) >= now);
-    }
+    DocStoreTest.testIgnoreCaseLikeQuery(datastore, DocStoreTest.MONGO_STORE);
```
Comment (kotharironak) [inline document-store/src/integrationTest/java/org/hypertrace/core/documentstore/mongo/MongoDocStoreTest.java:127]: - Can we mark this common test as `@Test` in DocStoreTest?
- And in the DocStoreTest.testIgnoreCaseLikeQuery, can we have for loop for each data-store?

So, this way, all the common tests will be moved to DocStoreTest.

File: document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresDatastore.java [hunk 2]
```diff
@@ -91,12 +92,12 @@ public boolean createCollection(String collectionName, Map<String, String> optio
                 + ");",
             collectionName, ID, DOCUMENT, CREATED_AT, UPDATED_AT);
     try (PreparedStatement preparedStatement = client.prepareStatement(createTableSQL)) {
-      int result = preparedStatement.executeUpdate();
-      return result >= 0;
+      preparedStatement.executeUpdate();
```
Comment (kotharironak) [inline document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresDatastore.java:95]: Can the result be `-1`? I am not sure if this case.

File: document-store/src/integrationTest/java/org/hypertrace/core/documentstore/postgres/PostgresDocStoreTest.java [hunk 3]
```diff
@@ -313,750 +201,61 @@ public void testDrop() {
...
-
-    Query query = new Query();
-    query.setFilter(new Filter(Filter.Op.IN, "size", inArray));
-    Iterator<Document> results = collection.search(query);
-    List<Document> documents = new ArrayList<>();
-    while (results.hasNext()) {
-      documents.add(results.next());
-    }
-    Assertions.assertEquals(documents.size(), 1);
+    DocStoreTest.testNotInQueryWithNumberField(datastore);
```
Comment (kotharironak) [inline document-store/src/integrationTest/java/org/hypertrace/core/documentstore/postgres/PostgresDocStoreTest.java:214]: This test doesn't exit for MongoDocStoreTest, can you enable it and see if works? If, not, don't move it to DocstoreTest. Let's keep those tests in DocStoreTest which works against both.

Comment (rish691) [reply document-store/src/integrationTest/java/org/hypertrace/core/documentstore/postgres/PostgresDocStoreTest.java:214]: I have added it for Mongo as well 
https://github.com/hypertrace/document-store/pull/41/files#diff-fab258440918057cd0bc0d16483437cc7dbfb88df8e6256b4efb55a763bb9712R538

File: document-store/src/integrationTest/java/org/hypertrace/core/documentstore/mongo/MongoDocStoreTest.java (hunk 1 above)
Comment (rish691) [reply document-store/src/integrationTest/java/org/hypertrace/core/documentstore/mongo/MongoDocStoreTest.java:127]: Will do it in next pr.

Comment (kotharironak) [reply document-store/src/integrationTest/java/org/hypertrace/core/documentstore/mongo/MongoDocStoreTest.java:127]: ok, let's see if possible to parametrized the test with document-store input.

**Review by kotharironak — APPROVED** [review]

**Pull request merged**